src/server/game/Battlegrounds/BattlegroundMgr.cpp

```diff
@@ -300,6 +301,16 @@ BattlegroundScriptTemplate const* BattlegroundMgr::FindBattlegroundScriptTemplat
     return Trinity::Containers::MapGetValuePtr(_battlegroundScriptTemplates, { mapId, BATTLEGROUND_TYPE_NONE });
 }
 
+void BattlegroundMgr::QueuePlayerForArena(Player const* player, uint8 teamSize, uint8 roles)
+{
+    std::unique_ptr<WorldPacket> worldPacket = std::make_unique<WorldPacket>(CMSG_BATTLEMASTER_JOIN_ARENA);
```

Comment:
This function is bad

* WorldPacket passed to BattlemasterJoinArena does not need to be heap allocated
* You don't need to fill it to call Read, just assign fields directly (this is extremely bad in case structure changes in future)